### PR TITLE
v1.9.3: defining cloud-init scripts in external files + jinja2 templates

### PR DESCRIPTION
**Pull request by jrzeszutek**
Added a possibility of using external scripts instead of writing the whole script in the blueprint. Moreover, added a possibility to use jinja2 templates.

### CODE REVIEW

**Comment by jrzeszutek** [comment]
Hi,

I've implemented the requested changes. Finally, I've decided to use `dict` type. I wasn't able to use these tags you suggested. I mean `!!python/tuple` and `!!python/dict`. Cloudify intristic functions like `get_input` weren't working in this case. I tried to work around it by using `concat` function, but then python interpreted it as a simple string and it wasn't working as well. But I've managed to get rid of additional properties like `external_content` and `params`, so I suppose it's better now.

File: cloudify_cloudinit/__init__.py [hunk 1]
```diff
@@ -25,15 +26,31 @@ def __init__(self, operation_inputs):
...
+                if 'file_resource' == resource_type:
+                    f['content'] = ctx.get_resource_and_render(resource_name, template_variables)
+            except ValueError:
+                ctx.logger.debug('No external resource recognized.')
+                pass
+        return config
 
     @staticmethod
-    def get_config(inputs):
+    def get_config(self, inputs):
```
Comment (0lvin) [inline cloudify_cloudinit/__init__.py]: Convert `get_config` to instance method as you also used class instance inside.

File: cloudify_cloudinit/__init__.py [hunk 2]
```diff
@@ -25,15 +26,30 @@ def __init__(self, operation_inputs):
         :param operation_inputs: The inputs from the operation.
         """
 
-        self.config = self.get_config(operation_inputs)
+        self.config = self.get_config(self, operation_inputs)
```
Comment (0lvin) [inline cloudify_cloudinit/__init__.py]: Convert:
```python
self.config = self.get_config(self, operation_inputs)
```
to
```python
self.config = self.get_config(operation_inputs)
```

And update unittests.

**Comment by 0lvin** [comment]
Implemented as part of #183

**Pull request closed without merging**